fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LucenePartitioner.java

```diff
@@ -579,7 +644,7 @@ public LuceneRecordCursor getOldestNDocuments(@Nonnull final LucenePartitionInfo
         LuceneScanParameters scan = new LuceneScanQueryParameters(
                 comparisons,
                 new LuceneQuerySearchClause(LuceneQueryType.QUERY, "*:*", false),
-                new Sort(new SortField(partitionTimestampFieldName, SortField.Type.LONG, false)),
+                new Sort(new SortField(partitionTimestampFieldName.replace('.', '_'), SortField.Type.LONG, false)),
```

Comment:
Without this, every document was missing the `partitionTimestampFieldName`, and thus the order they come back was seemingly random (maybe insertion order, or dependent on some merging).
For example, the tests use `complex.timestamp` which as a lucene field is `complex_timestamp`.
We may want to reconsider how we declare the partitioning timestamp.